plugins/inputs/sql/sql.go

```diff
@@ -0,0 +1,501 @@
+			if q.MeasurementColumn != "" && name == q.MeasurementColumn {
+				var ok bool
+				if measurement, ok = columnData[i].(string); !ok {
+					err := fmt.Errorf("measurement column type \"%T\" unsupported", columnData[i])
+					return err, len(columnNames), 0
+				}
+			}
+
+			if q.TimeColumn != "" && name == q.TimeColumn {
+				var e error
+
+				switch v := columnData[i].(type) {
+				case time.Time:
+					timestamp, e = v, nil
+				case int:
+					timestamp, e = internal.ParseTimestamp(q.TimeFormat, int64(v), "")
+				case int32:
+					timestamp, e = internal.ParseTimestamp(q.TimeFormat, int64(v), "")
+				case int64:
+					timestamp, e = internal.ParseTimestamp(q.TimeFormat, v, "")
+				case float32:
+					timestamp, e = internal.ParseTimestamp(q.TimeFormat, float64(v), "")
+				case float64:
+					timestamp, e = internal.ParseTimestamp(q.TimeFormat, v, "")
+				case string:
+					timestamp, e = internal.ParseTimestamp(q.TimeFormat, v, "")
+				default:
+					e = fmt.Errorf("column type \"%T\" unsupported", columnData[i])
+				}
+				if e != nil {
+					err := fmt.Errorf("parsing time failed: %v", e)
+					return err, len(columnNames), 0
+				}
+			}
+
+			if q.tagFilter.Match(name) {
+				var tagvalue string
+				switch v := columnData[i].(type) {
+				case string:
+					tagvalue = v
+				case []byte:
+					tagvalue = string(v)
+				case int:
+					tagvalue = strconv.FormatInt(int64(v), 10)
+				case int8:
+					tagvalue = strconv.FormatInt(int64(v), 10)
+				case int16:
+					tagvalue = strconv.FormatInt(int64(v), 10)
+				case int32:
+					tagvalue = strconv.FormatInt(int64(v), 10)
+				case int64:
+					tagvalue = strconv.FormatInt(int64(v), 10)
+				case uint:
+					tagvalue = strconv.FormatUint(uint64(v), 10)
+				case uint8:
+					tagvalue = strconv.FormatUint(uint64(v), 10)
+				case uint16:
+					tagvalue = strconv.FormatUint(uint64(v), 10)
+				case uint32:
+					tagvalue = strconv.FormatUint(uint64(v), 10)
+				case uint64:
+					tagvalue = strconv.FormatUint(uint64(v), 10)
+				case float32:
+					tagvalue = strconv.FormatFloat(float64(v), 'f', -1, 32)
+				case float64:
+					tagvalue = strconv.FormatFloat(float64(v), 'f', -1, 64)
+				case bool:
+					tagvalue = strconv.FormatBool(v)
+				case time.Time:
+					tagvalue = v.String()
+				case nil:
+					tagvalue = ""
+				default:
+					err := fmt.Errorf("tag column %q of type \"%T\" unsupported", name, columnData[i])
+					return err, len(columnNames), 0
+				}
+				if v := strings.TrimSpace(tagvalue); v != "" {
+					tags[name] = v
+				}
+			}
+
+			if q.fieldFilter.Match(name) {
+				var fieldvalue interface{}
+				switch v := columnData[i].(type) {
+				case string:
+					fieldvalue = v
+				case []byte:
+					fieldvalue = string(v)
+				case int:
+					fieldvalue = int64(v)
+				case int8:
+					fieldvalue = int64(v)
+				case int16:
+					fieldvalue = int64(v)
+				case int32:
+					fieldvalue = int64(v)
+				case int64:
+					fieldvalue = int64(v)
```

Comment:
I think these can be combined.

Reply:
If I combine those I get: `plugins/inputs/sql/sql.go:224:24: cannot convert v (type interface {}) to type int64: need type assertion`

Reply:
I have the same thing. I don’t know why it is like that.

Reply:
@ssoroka can I mark this as resolved or do you have any other idea on how to shorten this? As I've seen this quite often in the code it might be a common problem we should solve with an internal function!?